spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/PulsarConsumerTestUtil.java

```diff
@@ -53,6 +55,41 @@ public class PulsarConsumerTestUtil<T> implements TopicSpec<T>, SchemaSpec<T>, C
 
 	private List<String> topics;
 
+	private boolean calledUntil = false;
+
+	public static <T> TopicSpec<T> consumeMessages() throws PulsarClientException {
+		PulsarClient pulsarClient;
+		try {
+			pulsarClient = PulsarClient.builder().serviceUrl(
+				"pulsar://localhost:6650"
+			).build();
+		}
+		catch (PulsarClientException e) {
+			pulsarClient = PulsarClient.builder().serviceUrl(
+				PulsarTestContainerSupport.getPulsarBrokerUrl()
+			).build();
+		}
+		PulsarConsumerFactory<T> pulsarConsumerFactory = new DefaultPulsarConsumerFactory<>(
+			pulsarClient, List.of()
+		);
+		return new PulsarConsumerTestUtil<>(pulsarConsumerFactory);
+	}
+
+	public static <T> TopicSpec<T> consumeMessages(String url) throws PulsarClientException {
```

Comment:
This is great stuff @KartikShrivastava !

I am going to include a few points in this comment.

## Input validation
Let's add assert validate checks on each newly added method (consistent w/ the existing consumer factory variant.

## Checked exceptions
We (@jonas-grgt did all the work 😄 ) recently removed checked exceptions from our top level APIs. I think we should do the same here. The typical pattern is to `try/catch(PulsarClientException) -> throw PulsarException`. There is an example in one of the sample code blocks below. 

## Method chaining
I think we can chain these calls to consolidate them down a bit. E.g. the url variant can pass the client into the client variant etc. and we can do the same for the no-arg variant. Something like:

```java
public static <T> TopicSpec<T> consumeMessages(String url) {
	Assert.notNull(url, "url must not be null");
	try {
		return PulsarConsumerTestUtil.consumeMessages(PulsarClient.builder().serviceUrl(url).build());
	}
	catch (PulsarClientException ex) {
		throw new PulsarException(ex);
	}
}

public static <T> TopicSpec<T> consumeMessages(PulsarClient pulsarClient) {
	Assert.notNull(pulsarClient, "pulsarClient must not be null");
	var pulsarConsumerFactory = new DefaultPulsarConsumerFactory<T>(pulsarClient, List.of());
	return PulsarConsumerTestUtil.consumeMessages(pulsarConsumerFactory);
}
```
Wdyt?

Reply:
Sounds good to me, will update